src/main/java/com/emc/pravega/perf/PravegaTransactionWriterWorker.java

```diff
@@ -34,30 +34,33 @@
     PravegaTransactionWriterWorker(int sensorId, int eventsPerSec,
                                    int secondsToRun, boolean isRandomKey,
                                    int messageSize, Instant start,
-                                   ClientFactory factory, PerfStats stats,
-                                   String streamName,
-                                   long totalEvents,
+                                   PerfStats stats, String streamName,
+                                   long totalEvents, ClientFactory factory,
                                    int transactionsPerCommit) {
 
         super(sensorId, eventsPerSec, secondsToRun, isRandomKey,
-                messageSize, start, factory, stats, streamName, totalEvents);
+                messageSize, start, stats, streamName, totalEvents, factory);
 
         this.transactionsPerCommit = transactionsPerCommit;
         eventCount = new AtomicInteger(0);
         transaction = new AtomicReference<Transaction<String>>(producer.beginTxn());
     }
 
     @Override
-    public CompletableFuture writeData(String key, String data) throws TxnFailedException, IllegalStateException {
+    public CompletableFuture writeData(String key, String data) throws IllegalStateException {
```

Comment:
IllegalStateException is unchecked.

Reply:
same as above, since the writedata is abstract method of base class WriterWorker  , this class does not know TxnFailedException of pravega.

Reply:
yes, but IllegalState is unchecked. I does not need to be on this line. You can still throw it, there is just no need to type it here.

Reply:
i dint get this.. it throws IllegalStateException  correct?

Reply:
The method can throw IllegalStateException without it being declared in the function signature.
https://www.geeksforgeeks.org/checked-vs-unchecked-exceptions-in-java/

Reply:
done